pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -75,13 +75,68 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
 		return false, fmt.Errorf("cast of cp.Value failed: %v", okCpCast)
 	}
 
+	merged = false
+
+	// First merge transport socket matches with the same name
+	if (len(c.GetTransportSocketMatches())) > 0 && (len(cpValueCast.GetTransportSocketMatches()) > 0) {
+		for _, cpTsm := range cpValueCast.GetTransportSocketMatches() {
+			if cpTsm.GetName() == "" || (cpTsm.GetTransportSocket() == nil && cpTsm.GetMatch() == nil) {
+				// Don't try merging if TransportSocketMatches' TransportSocket and Match are both null or
+				// name is empty - name is used to match TransportSockets in patch to the one in live Cluster and
+				// if TransportSocket and Match are both null there's nothing to patch
+				continue
+			}
+
+			for _, tsm := range c.GetTransportSocketMatches() {
+				if tsm.GetName() != cpTsm.GetName() {
+					// Skip unmatched names
+					continue
+				}
+
+				if cpTsm.GetMatch() != nil {
+					// Replace Match if defined in patch
+					tsm.Match = cpTsm.Match
```

Comment:
NOt sure if this is right

Reply:
We are not stopping merging on this, for is still running ;)

Reply:
I guess if they explicitly set the match in the patch it seems ok? But usually they should not set the match probably.

Reply:
This is replacing only if explicitly set in patch (nil check) and I already see some usecases for Match patching.
eg. When I want default TSM to handle part of the traffic, but I add another one after that which handles rest of it.